### PR TITLE
Adding offline support for the map fragment and testing

### PR DESCRIPTION
**Pull request by aelalamy42**
This PR will add offline support to the map fragment.
photo1683232098

### CODE REVIEW

**Review by lucamouchel — COMMENTED** [review]
Hi, everything looks good to me, i just had a few questions.

what happens if the user is on the map fragment and turns on the wifi while on the map fragment? it seems like you are dealing with this in the onResume but this means a user has to change fragments then reopen the map to display the map.

Then i also saw you implemented the function is network available. I did that too but I'll use yours if we merge your PR first.

Otherwise everything lgtm

File: app/src/main/java/ch/epfl/culturequest/ui/map/MapsFragment.java [hunk 1]
```diff
@@ -96,15 +102,28 @@ public class MapsFragment extends Fragment {
...
 
+    private void checkInternet(){
+        isWifiAvailable = true;
+        this.getParentFragmentManager().beginTransaction().hide(unavailableFragment).show(this).setReorderingAllowed(true).commit();
+        if(!AndroidUtils.isNetworkAvailable()) {
+            isWifiAvailable = false;
+            Bundle args = new Bundle();
+            args.putBoolean("no_wifi", true);
+            this.getParentFragmentManager().beginTransaction().hide(this).show(unavailableFragment).setReorderingAllowed(true).commit();
+            Toast.makeText(getContext(), "It seems that you are not connected to the internet.", Toast.LENGTH_SHORT).show();
```
[the remark this answers is not in the document]
Comment (aelalamy42) [reply app/src/main/java/ch/epfl/culturequest/ui/map/MapsFragment.java]: Indeed! Thanks :)

File: app/src/main/java/ch/epfl/culturequest/ui/map/MapsFragment.java [hunk 2]
```diff
@@ -96,15 +102,28 @@ public class MapsFragment extends Fragment {
...
         }
     };
 
+    private void checkInternet(){
+        isWifiAvailable = true;
+        this.getParentFragmentManager().beginTransaction().hide(unavailableFragment).show(this).setReorderingAllowed(true).commit();
+        if(!AndroidUtils.isNetworkAvailable()) {
+            isWifiAvailable = false;
+            Bundle args = new Bundle();
+            args.putBoolean("no_wifi", true);
```
[the remark this answers is not in the document]
Comment (aelalamy42) [reply app/src/main/java/ch/epfl/culturequest/ui/map/MapsFragment.java]: Some left over code, thanks for pointing it out!

**Comment by aelalamy42** [comment]
Hi @lucamouchel, thanks for your review and your questions!
When the network is unavailable while on the already loaded maps fragment, it will just keep what was already fetched displayed. and then when changing fragments and going back the offline display will be shown.

**Comment by TheGuyWithoutH** [comment]
Is it possible to improve the coverage on what you added?

**Review by lucamouchel — APPROVED** [review]

**Pull request merged**